smart-contracts/assembly/contracts/FT/token-burn.ts

```diff
@@ -45,16 +40,17 @@ export function burn(binaryArgs: StaticArray<u8>): void {
  * @param amount -
  * @returns true if tokens has been burned
  */
-function _burn(addressToBurn: Address, amount: u64): boolean {
+export function _burn(addressToBurn: Address, amount: u64): void {
   const oldRecipientBalance = _balance(addressToBurn);
   const newRecipientBalance = oldRecipientBalance - amount;
 
   // Check underflow
-  if (oldRecipientBalance < newRecipientBalance) {
-    return false;
-  }
+  assert(
+    oldRecipientBalance > newRecipientBalance,
```

Comment:
opposite of < is >=

```suggestion
    oldRecipientBalance >= newRecipientBalance,
```

Reply:
i know but oldRecipientBalance = newRecipientBalance its not an overflow (meaning amount is 0)
IDK any reason for this to happen...